Source/WebCore/page/SecurityOrigin.h

```diff
@@ -249,7 +253,7 @@ template<class Encoder> inline void SecurityOrigin::encode(Encoder& encoder) con
     encoder << m_data;
     encoder << m_domain;
     encoder << m_filePath;
-    encoder << m_isUnique;
+    encoder << (m_uniqueOriginIdentifier ? std::optional { m_uniqueOriginIdentifier } : std::nullopt);
```

Comment:
No, we no longer allow IPC encoding / decoding invalid/empty ObjectIdentifiers. We explicitly fail decoding upon encountering such an identifier. The reason is security bugs. Recipients often failed to check that the id was valid before looking it up in a HashMap.

I could have chosen to use a std::optional<> for the data member but it would have made SecurityOrigin even larger, without much benefit IMO.

Reply:
One way to deal with this, which might be more elegant to use, is to define a distinct type for identifiers that can be empty, which can serialize differently. The underlying representation can still be the same. This is the exact relationship between `Ref` and `RefPtr`.

Reply:
Ok, the newer version uses Markable for the data member so we don't waste any space and I added an IPC encoder / decoder for Markable.